resources/tenants.py

```diff
@@ -36,16 +42,29 @@ def get(self, tenant_id=None):
 
 
     @admin_required
+    @jwt_required
     def post(self):
         data = Tenants.parser.parse_args()
         if TenantModel.find_by_first_and_last(data["firstName"], data["lastName"]):
             return { 'message': 'A tenant with this first and last name already exists'}, 401
 
         tenantEntry = TenantModel(**data) 
-        
         TenantModel.save_to_db(tenantEntry)
 
-        return tenantEntry.json(), 201
+        returnData = tenantEntry.json()
+
+        leaseData = request.json
+        leaseData.update({'tenantID': tenantEntry.id})
+
+        if ("occupants" in leaseData and "dateTimeEnd" in leaseData and "dateTimeStart" in leaseData and "propertyID" in leaseData):
```

Comment:
I think we'll need to ask the FrontEnd to send us nested Lease object in the JSON request when a lease needs to be created too so that we can avoid these checks here as it is confusing to what purpose they serve.

Reply:
occupants is not a required field, Let's remove that check, and leave the others for now.